lib/article_json/export/common/html/elements/base.rb

```diff
@@ -0,0 +1,85 @@
+              base.extend ClassMethods
+            end
+
+            # @param [ArticleJSON::Elements::Base] element
+            def initialize(element)
+              @element = element
+            end
+
+            # Export a HTML node out of the given element
+            # Dynamically looks up the right export-element-class, instantiates it
+            # and then calls the #build method.
+            # @return [Nokogiri::XML::NodeSet]
+            def export
+              exporter = is_base_class? ? self.class.build(@element) : self
+              exporter.export unless exporter.nil?
+            end
+
+            private
+
+            # Create a Nokogiri NodeSet wrapping a Node with the given `tag`
+            # @param [Symbol] tag - type of the created element
+            # @param [*Object] args - additional arguments for the element
+            # @yield [Nokogiri::XML::Element] Optional block can be passed, will
+            #                                 be executed with the generated
+            #                                 element as a parameter
+            # @return [Nokogiri::XML::NodeSet] Generated main element wrapped in a
+            #                                  NodeSet
+            def create_element(tag, *args)
+              document = Nokogiri::HTML.fragment('').document
+              element = document.create_element(tag.to_s, *args)
+              yield(element) if block_given?
+              wrap_in_node_set([element], document)
+            end
+
+            # Wrap a given list of Nokogiri Nodes in NodeSets
+            # @param [Array[Nokogiri::XML::Node]] elements
+            # @param [Nokogiri::HTML::Document] document
+            # @return [Nokogiri::XML::NodeSet]
+            def wrap_in_node_set(elements, document)
+              Nokogiri::XML::NodeSet.new(document).tap do |node_set|
+                elements.each { |element| node_set.push(element) }
+              end
+            end
+
+            # Check if the current instance is an instance of the base class or
+            # of a specific element. Since this common module is in a different
+            # namespace, a simple `self.class == Base` check does not work.
+            # @return [Boolean]
+            def is_base_class?
+              self.class.name.end_with?('::Base')
+            end
+
+            module ClassMethods
+              # Instantiate the correct sub class for a given element
+              # @param [ArticleJSON::Elements::Base] element
+              # @return [ArticleJSON::Export::Common::HTML::Elements::Base]
+              def build(element)
+                klass = exporter_by_type(element.type)
+                klass.new(element) unless klass.nil?
+              end
+
+              # Look up the correct exporter class based on the element type
+              # @param [Symbol] type
+              # @return [ArticleJSON::Export::Common::HTML::Elements::Base]
+              def exporter_by_type(type)
```

Comment:
The line breaks in this method feel weird. what about this?
```ruby
def exporter_by_type(type)
  key = type.to_sym
  custom_exporter = 
    ArticleJSON.configuration.element_exporter_for(export_format, key)
  custom_exporter || default_exporter_mapping[key]
end
```

Reply:
This also doesn't fit within the 80 character limit due to the 8 level deep indentation of this method..

Reply:
then maybe like this?
```ruby
def exporter_by_type(type)
  key = type.to_sym
  custom_exporter = ArticleJSON
    .configuration
    .element_exporter_for(export_format, key)
  custom_exporter || default_exporter_mapping[key]
end
```